### PR TITLE
no lib arch checks for mingw

### PR DESCRIPTION
**Pull request by autoantwort**
The cofffilereader fails when reading .a files with 'Could not find proper second linker member'

Fixes https://github.com/microsoft/vcpkg/issues/23450

### CODE REVIEW

**Review by strega-nil-ms — APPROVED** [review]
This is, imo, the correct solution. I'll merge once I get to work.

**Review by vicroms — APPROVED** [review]

File: include/vcpkg/base/util.h [hunk 1]
```diff
@@ -27,7 +27,9 @@ namespace vcpkg::Util
         template<class Vec, class Key>
         bool contains(const Vec& container, const Key& item)
         {
-            return std::find(container.begin(), container.end(), item) != container.end();
+            using std::begin;
```
Comment (BillyONeal) [inline include/vcpkg/base/util.h:30]: As an aside, this really doesn't belong in "vectors" anymore.

**Comment by AenBleidd** [comment]
As far as I see, there is no build artifacts, so the only way to test this is to build this manually?

**Comment by AenBleidd** [comment]
Ok, I tested this and it works fine now. Thanks for the quick fix, @autoantwort

**Comment by JackBoosY** [comment]
I think this PR caused https://github.com/microsoft/vcpkg/issues/23476. 
I can confirm that the build won't fails with [2022-03-08 Release](https://github.com/microsoft/vcpkg-tool/releases/tag/2022-03-08).

**Comment by dg0yt** [comment]
> I think this PR caused [microsoft/vcpkg#23476](https://github.com/microsoft/vcpkg/issues/23476).

I think this PR just fixed (enabled) the check. The new issues are real issues which existed before, but unnoticed.

**Comment by vicroms** [comment]
I agree with dg0yt. We'll need to fix regressions as we find them.

**Comment by JackBoosY** [comment]
Anyway, I hope we can make a fully test when our executable update.
And I hope we can enable all the checks in non-Windows in the future.

**Pull request merged**